node/src/Pos/Client/Txp/Util.hs

```diff
@@ -271,24 +254,48 @@ fixedToFee :: (MonadError TxError m, HasResolution a) => Fixed a -> m TxFee
 fixedToFee = integerToFee . ceiling
 
 data InputPickerState = InputPickerState
-    { _ipsMoneyLeft             :: !Coin
-    , _ipsAvailableOutputGroups :: ![UtxoGroup]
+    { _ipsMoneyLeft        :: !Coin
+    , _ipsAvailableOutputs :: !FlatUtxo
     }
 
 makeLenses ''InputPickerState
 
 type InputPicker = StateT InputPickerState (Either TxError)
 
+-- | Filters the input '[PendingTx]' to choose only the ones which are not
+-- yet persisted in the blockchain.
+nonConfirmedTransactions :: [PendingTx] -> [PendingTx]
+nonConfirmedTransactions = filter isPending
+  where
+    -- | Is this 'PendingTx' really pending?
+    isPending :: PendingTx -> Bool
+    isPending PendingTx{..} = case _ptxCond of
+        PtxInNewestBlocks _ -> False
+        PtxPersisted        -> False
+        _                   -> True
+
+-- | Returns the full list of "pending addresses", which are @output@ addresses
+-- associated to transactions not yet persisted in the blockchain.
+allPendingAddresses :: [PendingTx] -> Set Address
+allPendingAddresses = S.unions . map grabTxOutputs . nonConfirmedTransactions
```

Comment:
I think this is all just fine. This [patch](https://github.com/input-output-hk/cardano-sl/pull/2259) from vlad is reshuffling the code in different modules but the `HighThroughput` patch was definitely ported to `master`.